agent/api/container/status/managedagentstatus.go

```diff
@@ -0,0 +1,109 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package status
+
+import (
+	"errors"
+	"strings"
+)
+
+const (
+	// ManagedAgentStatusNone is the zero state of a managed agent
+	ManagedAgentStatusNone ManagedAgentStatus = iota
+	// ManagedAgentCreated represents a managed agent which has dependencies in place
+	ManagedAgentCreated
+	// ManagedAgentRunning represents a managed agent that has started sucessfully
+	ManagedAgentRunning
+	// ManagedAgentStopped represents a managed agent that has stopped
+	ManagedAgentStopped
+)
+
+// ManagedAgentStatus is an enumeration of valid states in the managed agent lifecycle
+type ManagedAgentStatus int32
+
+var managedAgentStatusMap = map[string]ManagedAgentStatus{
+	"NONE":    ManagedAgentStatusNone,
+	"CREATED": ManagedAgentCreated,
+	"RUNNING": ManagedAgentRunning,
+	"STOPPED": ManagedAgentStopped,
+}
+
+// String returns a human readable string representation of this object
+func (mas ManagedAgentStatus) String() string {
+	for k, v := range managedAgentStatusMap {
+		if v == mas {
+			return k
+		}
+	}
+	return "NONE"
+}
+
+// Terminal returns true if the managed agent status is STOPPED
+func (mas ManagedAgentStatus) Terminal() bool {
+	return mas == ManagedAgentStopped
+}
+
+// IsRunning returns true if the managed agent status is either RUNNING
+func (mas ManagedAgentStatus) IsRunning() bool {
+	return mas == ManagedAgentRunning
+}
+
+// BackendStatus maps the internal managedAgent status in the agent to that in the backend
+// This exists to force the status to be one of PENDING|RUNNING}|STOPPED
+func (mas ManagedAgentStatus) BackendStatus() string {
+	switch mas {
+	case ManagedAgentRunning:
+		return mas.String()
+	case ManagedAgentStopped:
+		return mas.String()
+	}
+	return "PENDING"
+}
+
+// ShouldReportToBackend returns true if the managedAgent status is recognized as a
+// valid state by ECS. Note that not all managedAgent statuses are recognized by ECS
+// or map to ECS states
+// For now we expect PENDING/RUNNING/STOPPED as valid states
+// we should only skip events that have ManagedAgentStatusNone
+func (mas ManagedAgentStatus) ShouldReportToBackend() bool {
+	return mas == ManagedAgentCreated || mas == ManagedAgentRunning || mas == ManagedAgentStopped
```

Comment:
1. `Created` status would map to the backend PENDING status and would mark the space between bind mount and actual execute command agent process start.
2. By this I'm assuming future managed agents would follow a similar bind mount/exec start process.
Since we're calling this a `ManagedAgent` I'm trying to adhere to that principle rather than hard-coding `executeCommandAgent` everywhere.  I don't foresee more such managedAgents, but there's a reason we decided to  define a generic structure, and this is why I've defined this status as such.
When we have another managedAgent, we should continue this conversation.

Reply:
1. I think my concern is more around introducing a new state, which might not be used for any purpose. 
It does indicate that bind mount is done, but how are we using the status, does it add any value? We could always know that execAgent is 'created' if the container has moved to 'created'. 
One place I can think this might help is during agent restarts, to know if execAgent is in  'created' state, but again irrespective of that, container status would already indicate that.

Reply:
also shouldn't we report only if its running or stopped? right now its reporting for Created as well (which backend is not aware of).

Reply:
For 2. I was thinking more around a generic ManagedAgent interface with methods like `ShouldReportToBackend` and each managed agent has its own implementation, kind of like our task resource model. But at this point, it might be an overkill. We can refactor when we add more managedAgents.